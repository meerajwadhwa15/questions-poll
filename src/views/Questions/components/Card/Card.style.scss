.card {
    background-color: $primary-background-color;
    padding: 10px;
    width: 300px;
    margin-right: 10px;
    margin-bottom: 15px;
    border: 1px solid $brand-black;
    color: $secondary-color;

    &__title {
        color: $brand-black;
    }

    &:hover {
        background-color: $secondary-background-color;
    }
}