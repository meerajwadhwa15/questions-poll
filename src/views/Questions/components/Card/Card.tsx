import React, { FunctionComponent } from "react";
import { Anchor } from "@/components";
import { question } from "@/types/questions";
import { dateFormater } from "@/utils";

export type Props = {
  question: question;
};

const List: FunctionComponent<Props> = ({
  question: { question, published_at, choices, url }
}) => (
  <div className="card">
    <Anchor to={url} title={question} rel="noopener">
      <h3 className="card__title ellipsis">{question}</h3>
      <p>Published Date: <strong>{dateFormater(published_at)}</strong></p>
      <p>Choices: <strong>{choices.length || 0}</strong></p>
    </Anchor>
  </div>
);

export default List;
