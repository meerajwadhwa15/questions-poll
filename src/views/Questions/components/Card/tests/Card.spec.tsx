import React from "react";
import Card from "../Card";
import { create } from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";

const question = {
  question: "Favourite programming language?",
  published_at: "2020-03-12T12:42:40.310135+00:00",
  url: "/questions/1",
  choices: [
    {
      choice: "Swift",
      votes: 39,
      url: "/questions/1/choices/1"
    }
  ]
};

const props = {
  question
};

test("Matches the snapshot with Questions Card", () => {
  const component = create(
    <BrowserRouter>
      <Card {...props} />
    </BrowserRouter>
  );
  expect(component.toJSON()).toMatchSnapshot();
});
