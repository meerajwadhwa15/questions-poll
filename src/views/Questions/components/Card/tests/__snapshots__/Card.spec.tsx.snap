// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches the snapshot with Questions Card 1`] = `
<div
  className="card"
>
  <a
    href="/questions/1"
    onClick={[Function]}
    rel="noopener"
    title="Favourite programming language?"
  >
    <h3
      className="card__title ellipsis"
    >
      Favourite programming language?
    </h3>
    <p>
      Published Date: 
      <strong>
        12/Jun/2020
      </strong>
    </p>
    <p>
      Choices: 
      <strong>
        1
      </strong>
    </p>
  </a>
</div>
`;
